feat(KL-104/상품피드페이지): create feed page

seoulyego opened the pull request:
## 📌 연관된 이슈
> 지라 이슈 링크
[KL-104: 상품피드페이지](https://ohhamma.atlassian.net/browse/KL-104)
## 📝 작업 내용
> 이번 PR에서 작업한 내용을 간략히 설명해주세요(이미지 첨부 가능)

### 피드 페이지 작성
- 이미지 썸네일
  세상에서 가장 예쁜 아이들로 보여드렸습니다.
  
- 기본 필터 사이드바
  국가, 도시, 카테고리 및 서브 카테고리를 선택할 수 있습니다.
  
- 추가 필터
  추가 필터를 선택할 수 있습니다.
  
- 선택 필드
  기본 필터, 추가 필터에서 설정한 피드가 표시됩니다.
  x 버튼을 눌러 선택된 필드를 제거할 수 있습니다.
  초기화 버튼을 눌러 선택된 모든 필드를 제거할 수 있습니다.
  정렬 버튼을 만들어만 두었습니다.
  
- 리뷰 리스트
  컨텐츠 미리보기를 9개까지 보여줍니다.
  로딩 시에는 로딩 컴포넌트를 표시합니다.

- 라우터 추가
  Feed Page를 라우터에 추가하였습니다.

- 컨플릭트 처리
  컨플릭트 해결 완.

## 🌳 작업 브랜치명
> 현재 작업중인 브랜치를 적어주세요
KL-104/FeedPage

## 📸 스크린샷 (선택)
![imgae](https://github.com/user-attachments/assets/5a9c2584-f2f0-4c98-b110-53f5e64fd44c)

## 💬 리뷰 요구사항 (선택)
> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Added a new `FeedPage` component with enhanced filtering functionality.
	- Introduced new filter components: `AdditionalFilter` and `FilterSection` for improved user interaction.
	- Implemented `useSelectedFilter` hooks for managing user selections across filters.
	- Created `ShowHideButton`, `IconTextButton`, and `LoadingContent` components for better user interface experience.
	- Implemented city selection functionality based on the selected country.

- **Bug Fixes**
	- Resolved issues related to import paths and component structure for better organization.

- **Refactor**
	- Simplified product data structure and modified related components to enhance maintainability.
	- Consolidated prop passing in components to streamline data management.

- **Style**
	- Updated styling files and created new styled components to enhance visual consistency and responsiveness across the application.
	- Introduced new animations and loading states to improve user experience during data fetching.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

seoulyego committed:
KL-113/chore: untrack node_modules

tracked file 목록에서 node_modules/를 삭제하였습니다.
더이상 node_modules를 tracking하지 않습니다.
.gitignore 목록에 node_modules가 추가되었습니다.

seoulyego committed:
KL-113/chore: untrack node_modules

tracked file 목록에서 node_modules/를 삭제하였습니다.
더이상 node_modules를 tracking하지 않습니다.
.gitignore 목록에 node_modules가 추가되었습니다.

seoulyego committed:
KL-113/refactor: seperate presentational component

LikeButton 컴포넌트의 비즈니스 로직과 UI 렌더를 분리하였습니다.
LikeButton은 비즈니스 로직만 담당하고 UI는 IconTextButton에서 렌더합니다.
이제 아이콘 버튼 또는 텍스트 버튼의 렌더링 컴포넌트를 재사용할 수 있습니다.

seoulyego committed:
KL-113/refactor: seperate presentational component

LikeButton 컴포넌트의 비즈니스 로직과 UI 렌더를 분리하였습니다.
LikeButton은 비즈니스 로직만 담당하고 UI는 IconTextButton에서 렌더합니다.
이제 아이콘 버튼 또는 텍스트 버튼의 렌더링 컴포넌트를 재사용할 수 있습니다.

seoulyego committed:
kL-113/feat: add ShowHideButton

숨기기 보여주기 버튼을 추가하였습니다.

seoulyego committed:
kL-113/feat: add ShowHideButton

숨기기 보여주기 버튼을 추가하였습니다.

seoulyego committed:
KL-113/feat: add BasicFilter Component

Basic Filter 컴포넌트를 추가하였습니다.
기능 구현 및 테스트가 완료되었습니다.

예정된 작업은 다음과 같습니다.
- Radio 버튼 및 그룹 적용 (Country Section)
- 체크박스 그룹 적용 (City Section, Category Section)
- Check all 체크박스 적용 (City Section, Category Section)
- 디자인 적용 (Basic Filter)
- API 연결

seoulyego committed:
KL-113/feat: add BasicFilter Component

Basic Filter 컴포넌트를 추가하였습니다.
기능 구현 및 테스트가 완료되었습니다.

예정된 작업은 다음과 같습니다.
- Radio 버튼 및 그룹 적용 (Country Section)
- 체크박스 그룹 적용 (City Section, Category Section)
- Check all 체크박스 적용 (City Section, Category Section)
- 디자인 적용 (Basic Filter)
- API 연결

seoulyego committed:
KL-113/chore: add package antd, use-axios-client

antd, use-axios-client 패키지를 추가하였습니다.

antd: React UI 라이브러리
use-axios-client: useAxios 훅을 사용하기 위한 라이브러리

seoulyego committed:
KL-113/chore: add package antd, use-axios-client

antd, use-axios-client 패키지를 추가하였습니다.

antd: React UI 라이브러리
use-axios-client: useAxios 훅을 사용하기 위한 라이브러리

seoulyego committed:
KL-113/rename: modify directory name (Filter -> BasicFilter)

- 디렉토리 이름을 컴포넌트 이름과 동일하게 변경하였습니다. (Filter -> BasicFilter)
- depth에 따른 styled-component 영역을 재지정하였습니다.
- BasicFilter 컴포넌트의 width를 고정하였습니다.

seoulyego committed:
KL-113/rename: modify directory name (Filter -> BasicFilter)

- 디렉토리 이름을 컴포넌트 이름과 동일하게 변경하였습니다. (Filter -> BasicFilter)
- depth에 따른 styled-component 영역을 재지정하였습니다.
- BasicFilter 컴포넌트의 width를 고정하였습니다.

seoulyego committed:
KL-104/chore: untracked node_modules/.cache

이제 node_modules/.cache를 추적하지 않습니다.

seoulyego committed:
KL-104/chore: untracked node_modules/.cache

이제 node_modules/.cache를 추적하지 않습니다.

seoulyego committed:
Merge branch 'KL-113/BasicFilter' into KL-104/FeedPage

seoulyego committed:
KL-114/refactor: divide custom hooks from context

- Context API 사용을 위한 context 파일과 custom hooks를 별도의 파일로 분리했습니다.
- src 아래에 contexts와 hooks디렉토리가 추가되었습니다.
- context 파일과 custom hooks의 경로가 각각 src/contexts, src/hooks로 변경되었습니다.
- BasicFilter에 사용된 context와 custom hooks의 import 경로를 변경하였습니다.

seoulyego committed:
KL-114/refactor: divide custom hooks from context

- Context API 사용을 위한 context 파일과 custom hooks를 별도의 파일로 분리했습니다.
- src 아래에 contexts와 hooks디렉토리가 추가되었습니다.
- context 파일과 custom hooks의 경로가 각각 src/contexts, src/hooks로 변경되었습니다.
- BasicFilter에 사용된 context와 custom hooks의 import 경로를 변경하였습니다.

seoulyego committed:
KL-114/chore: add ant-design/icons and modify setting for vscode extension

@ant-design/icons 패키지를 추가하였습니다.
px to rem 확장의 설정이 추가되었습니다.

seoulyego committed:
KL-114/chore: add ant-design/icons and modify setting for vscode extension

@ant-design/icons 패키지를 추가하였습니다.
px to rem 확장의 설정이 추가되었습니다.

seoulyego committed:
KL-114/feat: add Context, Provider and Hooks for Additional Filter

추가 필터를 위한 Context, Provider, Hooks가 추가되었습니다.
selectedFilter 상태를 통해 편의점, 고수 등의 필터를 저장할 수 있습니다.

seoulyego committed:
KL-114/feat: add Context, Provider and Hooks for Additional Filter

추가 필터를 위한 Context, Provider, Hooks가 추가되었습니다.
selectedFilter 상태를 통해 편의점, 고수 등의 필터를 저장할 수 있습니다.

seoulyego committed:
KL-114/feat: add Additional Filter

추가 필터 기능이 추가되었습니다.
이제 편의점, 고수 등의 추가 필터를 Toogle로 켜고 끌 수 있습니다.

seoulyego committed:
KL-114/feat: add Additional Filter

추가 필터 기능이 추가되었습니다.
이제 편의점, 고수 등의 추가 필터를 Toogle로 켜고 끌 수 있습니다.

seoulyego committed:
KL-114/design: add margins at preview

PreviewContent에서 보여주는 정보에 별점을 추가하였습니다.
Preview style에 마진을 조정했습니다.
PreviewContent의 props로 분리 전달하던 Product 속성을 객체 형태로 전달하도록 수정하였습니다.

seoulyego committed:
KL-114/design: add margins at preview

PreviewContent에서 보여주는 정보에 별점을 추가하였습니다.
Preview style에 마진을 조정했습니다.
PreviewContent의 props로 분리 전달하던 Product 속성을 객체 형태로 전달하도록 수정하였습니다.

seoulyego committed:
KL-114/style: modify if state in handleLiked function

handleLiked 함수의 if 분기문을 수정하였습니다.

seoulyego committed:
KL-114/style: modify if state in handleLiked function

handleLiked 함수의 if 분기문을 수정하였습니다.

seoulyego committed:
KL-114/design: modify the placement of elements

기본 필터의 요소들에 대해 배치를 수정하였습니다.

seoulyego committed:
KL-114/design: modify the placement of elements

기본 필터의 요소들에 대해 배치를 수정하였습니다.

seoulyego committed:
KL-104/feat: add Feed Page

Feed Page를 추가하였습니다.
기본 필터, 추가 필터, 리뷰 미리보기가 포함되어 있습니다.

예정된 작업
- 선택 필드 추가
- 페이지 디자인

seoulyego committed:
KL-104/feat: add Feed Page

Feed Page를 추가하였습니다.
기본 필터, 추가 필터, 리뷰 미리보기가 포함되어 있습니다.

예정된 작업
- 선택 필드 추가
- 페이지 디자인

seoulyego committed:
KL-115/chore: add code spell word

code spell 단어에 antd가 추가되었습니다.

seoulyego committed:
KL-115/chore: add code spell word

code spell 단어에 antd가 추가되었습니다.

seoulyego committed:
KL-115/refactor: modify tags

tags를 styled-component로 유지하도록 변경하였습니다.

seoulyego committed:
KL-115/refactor: modify tags

tags를 styled-component로 유지하도록 변경하였습니다.

seoulyego committed:
KL-115/refactor: apply BlueTag

기존의 Tag 컴포넌트 대신 BlueTag styled-component를 적용하였습니다.

seoulyego committed:
KL-115/refactor: apply BlueTag

기존의 Tag 컴포넌트 대신 BlueTag styled-component를 적용하였습니다.

seoulyego committed:
KL-115/refactor: move ContextProvider from each module to the page

각 모듈에 존재하던 ContextProvider를 FeedPage로 옮겼습니다.
이제 단일화된 전역 상태 관리가 가능합니다.

seoulyego committed:
KL-115/refactor: move ContextProvider from each module to the page

각 모듈에 존재하던 ContextProvider를 FeedPage로 옮겼습니다.
이제 단일화된 전역 상태 관리가 가능합니다.

seoulyego committed:
KL-115/refactor: modify empty check for Object

selectedCountry 상태에 대한 빈 객체 확인 코드에 대한 수정

seoulyego committed:
KL-115/refactor: modify empty check for Object

selectedCountry 상태에 대한 빈 객체 확인 코드에 대한 수정

seoulyego committed:
KL-115/design: fit contents to frame

- PreviewContent 컴포넌트에서 이미지 썸네일을 프레임에 채워서 출력합니다.
- BasicFilter의 너비가 내용 너비 이하로 줄어들지 않습니다.

seoulyego committed:
KL-115/design: fit contents to frame

- PreviewContent 컴포넌트에서 이미지 썸네일을 프레임에 채워서 출력합니다.
- BasicFilter의 너비가 내용 너비 이하로 줄어들지 않습니다.

seoulyego committed:
KL-115/feat: add SelectedField

SelectedField 컴포넌트가 추가되었습니다.
- 기본 필터, 추가 필터에서 선택된 필드가 표시됩니다.
- 선택되지 않은 필터는 전체로 표시됩니다.
- x 버튼을 누르면 선택된 필드가 제거됩니다.

seoulyego committed:
KL-115/feat: add SelectedField

SelectedField 컴포넌트가 추가되었습니다.
- 기본 필터, 추가 필터에서 선택된 필드가 표시됩니다.
- 선택되지 않은 필터는 전체로 표시됩니다.
- x 버튼을 누르면 선택된 필드가 제거됩니다.

seoulyego committed:
KL-116/refactor: minor update for components

주로 CSS 코드 변경
PreviewContent props 변경 (city -> country)
Field 용 Tag 컴포넌트 추가 (WhiteFieldTag, BlueFieldTag)

seoulyego committed:
KL-116/refactor: minor update for components

주로 CSS 코드 변경
PreviewContent props 변경 (city -> country)
Field 용 Tag 컴포넌트 추가 (WhiteFieldTag, BlueFieldTag)

seoulyego committed:
KL-116/refactor: modify custom hook

Custom Hook 이름 변경
useSelectedFilterProperty -> useSelectedFilter

seoulyego committed:
KL-116/refactor: modify custom hook

Custom Hook 이름 변경
useSelectedFilterProperty -> useSelectedFilter

seoulyego committed:
KL-116/design: make layout for FeedPage

FeedPage의 레이아웃과 디자인을 적용하였습니다.
FeedPage 디렉토리를 생성하였습니다.
기존 components 디렉토리에 있던 FeedPage의 하위 컴포넌트를 FeedPage 디렉토리로 옮겼습니다.

seoulyego committed:
KL-116/design: make layout for FeedPage

FeedPage의 레이아웃과 디자인을 적용하였습니다.
FeedPage 디렉토리를 생성하였습니다.
기존 components 디렉토리에 있던 FeedPage의 하위 컴포넌트를 FeedPage 디렉토리로 옮겼습니다.

seoulyego committed:
KL-104/feat: create Feed Page Thumbnail

Feed Page 상단에 이미지 썸네일을 추가하였습니다.

seoulyego committed:
KL-104/feat: create Feed Page Thumbnail

Feed Page 상단에 이미지 썸네일을 추가하였습니다.

seoulyego committed:
KL-104/rename: modify directory name style -> styles

- Feed Page의 썸네일 이미지의 위치를 조정하였습니다.
- 선택 필드의 초기화 버튼의 위치를 변경하였습니다.
- 추가 필터와 선택 필드의 title의 크기를 변경하였습니다.
- 추가 필터를 토글에서 버튼으로 변경하였습니다.
- style directory의 이름을 styles로 변경하였습니다.

seoulyego committed:
KL-104/chore: reset development environment

rebase 과정에서 발생한 개발 환경을 다시 세팅하였ㅅㅂ니다.

seoulyego committed:
KL-104/refactor: modify styles directory name

style 디렉토리의 import 경로를 수정하였습니다.

seoulyego committed:
Merge branch 'KL-104/FeedPage' of https://github.com/taco-official/klkl-client into KL-104/FeedPage

seoulyego committed:
KL-104/feat: add route for feed page

Feed Page을 라우터에 추가하였습니다.

seoulyego committed:
KL-104/design: remove inline-block display property in css

Preview Content의 display 속성에서 inline-block을 삭제하였습니다.

seoulyego committed:
KL-104/fix: modify filter button when reset selectedFilter

selectedFilter가 초기화될 때 default 버튼으로 변경되지 않는 버그를 수정하였습니다.

seoulyego committed:
Update src/pages/FeedPage/AdditionalFilter/FilterSection.jsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

seoulyego committed:
KL-104/design: modify background-color for PreviewContainer

PreviewContainer의 배경색을 변경하였습니다.

seoulyego committed:
Merge branch 'KL-104/FeedPage' of https://github.com/taco-official/klkl-client into KL-104/FeedPage

seoulyego committed:
KL-104/refactor: modify function initializeFilterState in FilterSection

- initializeFilterState에서 reduce의 초기값을 빈 배열로 변경하였습니다.
- acc.push 메소드를 사용하였습니다.

seoulyego committed:
Update src/pages/FeedPage/BasicFilter/CitySection.jsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

sayoon-mandarine commented on package.json:
antd 버전이 내려갔어

sayoonnn commented on package.json:
```diff
@@ -7,7 +7,7 @@
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
-    "antd": "^5.19.4",
+    "antd": "^5.19.3",
```
버전이 오히려 내려갔어요

seoulyego replied:
rebase 충돌을 처리하는 과정에서 버전 차이가 발견됐어요.
어느 브랜치가 19.3인지는 몰라도 사윤님이랑 버전 차이가 나는 것 같아서 
이전 버전으로 두는 것이 패키지 호환성 측면에서 좋을 것 같아서 19.3으로 유지했습니다.
근데 각 브랜치의 커밋 기록을 다시 살펴봤는데 5.19.3 버전이 설치된 시점이 파악되지 않네요.
기존대로 5.19.4로 변경해서 올려둘게요.

sayoonnn commented on src/components/Button/IconTextButton.jsx:
```diff
@@ -0,0 +1,29 @@
+  return (
+    <IconContext.Provider value={iconValue}>
+      <>
+        {Icon || null}
```
기본값이 null 이면 그냥 Icon만 넣어도 되지 않나요?
에러가 나나요?

seoulyego replied:
기본값이 null인 것을 미처 고려하지 못 했네요. Icon과 아래 text까지 수정해서 올려놓을게요.

sayoonnn commented on src/components/Button/ShowHideButton.jsx:
```diff
@@ -0,0 +1,33 @@
+  isOption = false,
+}) {
+  const iconAttr = useMemo(() => ({ onClick: handleClick }), [handleClick])
+  const iconValue = useMemo(
```
단순히 객체를 return 하는 경우에는 성능상 큰 차이도 없을 뿐 더러, 오히려 메모리에 탑재하는 오버헤드가 더 크다고 합니다

그리고 memorization을 하더라도, 상위 컴포넌트에서 state가 변경되어 재 렌더링 되는 경우는 그 하위 컴포넌트도 재 렌더링 되기 때문에 의미 없는 경우 같습니다

나중에 useCallback 하고 useMemo 같이 수정하도록 합니당

seoulyego replied:
추후 전체적으로 useCallback useMemo 죽이기 진행합니다.

sayoonnn commented on src/components/preview/PreviewContent.jsx:
```diff
@@ -1,77 +1,87 @@
   return (
     <PreviewContainer>
-      <ThumbnailContainer className="productThumbnail">
+      <ThumbnailContainer id="productThumbnail">
```
css 에서 접근하는 것 같진 않던데 id를 사용한 이유가 있을까요

seoulyego replied:
기존에 PreviewContent랑 LoadingContent에서 각 컨테이너들을 별도로 사용했는데,
동일하게 사용하는 것으로 바꾸면서 id 및 class로 필요한 속성을 구분하게 했어요.
LoadingContent 속성은 분리를 해놨는데 PreviewContent에 대한 속성을 지정을 안 해놨네요.
id 사용하는 것으로 css 변경해놓을게요.

근데 궁금한게 지정자가 간혹 안 먹는 경우가 있는데 왜 그런지 아시나요?
<img width="226" alt="image" src="https://github.com/user-attachments/assets/cdb1cb78-1031-47e2-90c9-d4d5e7496669">
<img width="222" alt="image" src="https://github.com/user-attachments/assets/2ec16cdc-cb04-435c-87fd-560e789b9160">

sayoonnn replied:
scss라서 그런지 & > # 이렇게 해야 붙을 때도 있더라구요

seoulyego replied:
앞에 &도 붙여봤는데 안 되더라구여...
지시자 앞에 타입을 정해주면 되긴 되던데
어차피 id는 하나 밖에 못 쓰는데 타입이 반드시 필요한가...

sayoonnn commented on src/contexts/AdditionalFilterContext.jsx:
useCallback 사용하셨지만,
selectedFilter state가 변경되었을때 어차피 그 하위 컴포넌트는 모두 rerender 되어서 의미 없는 것 같습니다
개발자 모드 깔아서 확인해봐도 동일하구요

어차피 Zustand 사용할 거라 이렇게 내려보낼 필요가 없어질 것 같아서 나중에 한꺼번에 고치시면 될 것 같습니다

seoulyego replied:
화긴 화긴

sayoonnn commented on src/pages/FeedPage/AdditionalFilter/AdditionalFilter.style.js:
```diff
@@ -0,0 +1,25 @@
+  }
+`
+
+const FilterContainer = styled(ContentContainer)``
```
나중에 적용하시려고 남겨둔건가용?

seoulyego replied:
column-gap이 ContentContainer에 적용되어 있었네요.
추가 필터랑 선택 필드에 column-gap row-gap 통일하고 FilterContainer, FieldContainer 둘 다 제거했습니다.

sayoonnn commented on src/pages/FeedPage/SelectedField/SortButton.jsx:
이부분은 아직 하고계신거죠?

위에 알림창도 안뜨게 하면 좋을 것 같습니다

seoulyego replied:
별점, 좋아요, 최신, 오래된 순으로 명칭 변경했어요.
알림창은 클릭했을 때 알림창 말인거죠? 제거해놨습니다.

sayoonnn reviewed (dismissed):
코멘트 확인해주시면 감사하겠습니당

seoulyego committed:
KL-104/chore: modify antd version to 5.19.4

antd 버전을 5.19.4로 변경하였습니다.

seoulyego committed:
KL-104/refactor: modify return value IconTextButton

IconTextButton의 리턴을 수정하였습니다.
- Icon과 text 공간을 wrapping하는 div를 추가하였습니다.
- Icon과 text의 기본값이 null인 것을 고려하여 코드를 변경하였습니다.

seoulyego committed:
KL-104/fix: Fix icon not rendering

Icon의 사용 방법이 잘못되어 렌더에 문제가 있었습니다.
정상적으로 렌더링 될 수 있도록 수정하였습니다.

seoulyego committed:
KL-104/design: use id for ThumbnailContainer

ThumbnailContainer의 id를 활용하여 단독 속성을 지정해주었습니다.

seoulyego committed:
KL-104/feat: pass thmbnail image src as props

PreviewContent의 thumbnail 이미지 주소를 props로 전달받습니다.
ProductList 더미 데이터에 thumnail 이미지 주소를 추가하였습니다.

seoulyego committed:
KL-104/fix: limit the number of tags in PreviewContent

미리보기의 태그가 4개 이상일 때 미리보기 영역을 넘어가는 버그가 발견되었습니다.
미리보기에서 표시하는 태그를 최대 4개로 제한됩니다.

seoulyego committed:
KL-104/refactor: unify css properties of AdditionalFilter and SelectedField

AdditionalFilter와 SelectedField의 css 속성을 통합하였습니다.
- FilterSection와 FieldSection의 속성을 StyledSection으로 통합
- FilterContainer와 FieldContainer의 gap 속성을 ContentContainer로 통합

SelectedField의 구조를 변경하였습니다.
- div 요소 변경
- SortButton에 align-self 적용

필요없는 컨테이너를 제거하였습니다.
- FilterSection 제거
- FilterContainer 제거
- FilterWrapper 제거
- FieldContainer 제거

seoulyego committed:
KL-104/test: modify tag names in dummy data

ProductList 더미 데이터의 tag 명을 변경하였습니다.

seoulyego committed:
KL-104/feat: change the dropdown menu label of SortButton

SortButton의 Dropdown 메뉴 레이블 변경했습니다.
클릭 시 상단에 출력되는 테스트 메세지를 제거하였습니다.

seoulyego committed:
Merge branch 'KL-104/FeedPage' of https://github.com/taco-official/klkl-client into KL-104/FeedPage

seoulyego committed:
KL-104/chore: change ECMA version to 2020 in eslint

eslint의 ECMA version을 ECMA Script2020으로 변경하였습니다.

seoulyego committed:
KL-104/style: apply optional chaining in CitySection

City 목록을 fetch하기 전 selectedCountry 검증 과정에 optional chaining을 적용하였습니다.
